azure/Kqlmagic/my_aad_helper.py

```diff
@@ -138,6 +138,12 @@ def acquire_token(self, **options):
             url = code[OAuth2DeviceCodeResponseParameters.VERIFICATION_URL]
             device_code = code[OAuth2DeviceCodeResponseParameters.USER_CODE].strip()
             
+
+            if options.get("login_code_destination") !="browser" or  options.get("notebook_app")=="papermill":
```

Comment:
the or  options.get("notebook_app")=="papermill" won't work, because it doesnot guarantee the or  login_code_destination is set to an email. Maybe the right thing to do, is if it is papermill and login_code_destination is set to browser, to return an error.